### PR TITLE
Migrate elasticsearch-exporter and stackdriver-exporter charts from stable repo

### LINKED ISSUE #133: Migrate stackdriver-exporter and elasticsearch-exporter
https://github.com/helm/charts/tree/master/stable/elasticsearch-exporter and https://github.com/helm/charts/tree/master/stable/stackdriver-exporter are pending migration according to https://github.com/helm/charts/issues/21103

They should be migrated into this repo

### PR DESCRIPTION
**Pull request by torstenwalter**
#### What this PR does / why we need it:

It migrated the existing `elasticsearch-exporter` and `stackdriver-exporter charts` from stable repository here.

Below are the steps which have been used to migrate the history:
```
# Prepare charts repository for merge
git clone git@github.com:helm/charts.git /tmp/charts
cd /tmp/charts
# Filter history for helm charts to reduce repository size
git filter-repo \
  --path-glob 'stable/elasticsearch-exporter/*' \
  --path-glob 'stable/stackdriver-exporter/*' \
  --path-rename stable/elasticsearch-exporter/:charts/prometheus-elasticsearch-exporter/ \
  --path-rename stable/stackdriver-exporter/:charts/prometheus-stackdriver-exporter/

# Merge history into helm-charts repository
git clone https://github.com/prometheus-community/helm-charts.git /tmp/prometheus-helm-charts
cd /tmp/prometheus-helm-charts
git checkout -b merge-exporter-charts
git remote add -f charts /tmp/charts/.git
git merge --allow-unrelated-histories -s ours --no-commit charts/master
git read-tree --prefix=charts/prometheus-elasticsearch-exporter/ -u charts/master:charts/prometheus-elasticsearch-exporter/
git read-tree --prefix=charts/prometheus-stackdriver-exporter/ -u charts/master:charts/prometheus-stackdriver-exporter/
git commit -s -S -m "Add elasticsearch-export and stackdriver-exporter charts"

```

Steps which have been done after merging the history:
    
- increased minor chart versions
- added maintainers to CODEOWNERS
- adjusted READMEs
- fixed markdown linting issues
- renamed charts to have `prometheus-` prefix

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #133 

#### Special notes for your reviewer:
I have not contacted the existing maintainers in advance so please ensure that we invite them to this repository before merging it.

@caarlos0 @desaintmartin @svenmueller for prometheus-elasticsearch-exporter
@apenney for prometheus-stackdriver-exporter

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (scottrigby) [inline charts/prometheus-elasticsearch-exporter/README.md:27]:
```suggestion
$ helm install [RELEASE_NAME] prometheus-community/prometheus-elasticsearch-exporter [flags]
```

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (scottrigby) [inline charts/prometheus-elasticsearch-exporter/README.md:30]:
```suggestion
$ helm install --name [RELEASE_NAME] prometheus-community/prometheus-elasticsearch-exporter [flags]
```

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (scottrigby) [inline charts/prometheus-elasticsearch-exporter/README.md:33]:
```suggestion
The command deploys Elasticsearch-Exporter on the Kubernetes cluster using the default configuration.
```

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (scottrigby) [inline charts/prometheus-elasticsearch-exporter/README.md:80]:
```suggestion
$ helm inspect values prometheus-community/prometheus-elasticsearch-exporter
```

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (scottrigby) [inline charts/prometheus-elasticsearch-exporter/README.md:83]:
```suggestion
$ helm show values prometheus-community/prometheus-elasticsearch-exporter
```

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (scottrigby) [inline charts/prometheus-elasticsearch-exporter/README.md:71]: Are these applicable to elasticsearch-exporter or stackdriver-exporter? Just checking due to the other typos, and can't immediately tell since I'm not familiar with these charts.

**Review by scottrigby — CHANGES_REQUESTED** [review]
Just a few suggested changes, and a question.

Apart from that:
- I added "Fixes #133" to the PR description
- We should get confirmation from the maintainers of these charts from stable. If we do, add them per the PROCESSES doc (PS, the plan now is to update the PROCESSES doc for adding new maintainers with this step:
    > To add people, file an issue on `https://github.com/prometheus-community/community`, with "add @x to team helm-charts/helm-charts-maintainers`. This link should work: https://github.com/prometheus-community/helm-charts/issues/new?title=New+maintainer+request+%5Bgithub+username%5D+issue+title&body=@prometheus-community/helm-charts-admins

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (torstenwalter) [reply charts/prometheus-elasticsearch-exporter/README.md:27]: I think this was there by intention as deploying the chart without it does not make sense.

**Comment by monotek** [comment]
We use a stack driver exporter too at kiwigrid: https://github.com/kiwigrid/helm-charts/tree/master/charts/stackdriver-exporter

We were thinking about to move it here too. Maybe you'd like to compare? If you choose to use this one we likely will deprecate ours.

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (torstenwalter) [reply charts/prometheus-elasticsearch-exporter/README.md]: Was to late yesterday ;-) It's fixed now.

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (torstenwalter) [reply charts/prometheus-elasticsearch-exporter/README.md:71]: This is ok:
https://github.com/helm/charts/tree/master/stable/elasticsearch-exporter#to-200
https://github.com/helm/charts/tree/master/stable/elasticsearch-exporter#to-300

**Comment by torstenwalter** [comment]
@monotek I think you would be in a better position to compare both. I actually never used both. Just trying to help migrating stuff from stable repo so that it does not get lost.

**Comment by torstenwalter** [comment]
@monotek The ci job is failing for the stackdriver-exporter:

```
==> Logs of container prometheus-stackdriver-exporter-qnfc2o30bo-786c54f9f9-mc7zv
------------------------------------------------------------------------------------------------------------------------
time="2020-10-07T06:55:25Z" level=info msg="Starting stackdriver_exporter (version=0.6.0, branch=HEAD, revision=59b7f38fa3d7e7ff8fd7134dd373c91051d5c16c)" source="stackdriver_exporter.go:121"
time="2020-10-07T06:55:25Z" level=info msg="Build context (go=go1.11.2, user=root@28c8704f4734, date=20181202-06:28:44)" source="stackdriver_exporter.go:122"
time="2020-10-07T06:55:25Z" level=error msg="Error creating Google client: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information." source="stackdriver_exporter.go:126"
```

Do you know if credentials have been available on the stable repository? I am wondering how it worked over there.
Do you have any idea how we could fix the ci build?

**Comment by desaintmartin** [comment]
Thanks for the initiative, of course i'm willing to continue maintaining it here!
Note that I am already maintainer of other chart here so no need to add me to the org.

**Comment by torstenwalter** [comment]
@desaintmartin Could you review changes for that chart?

File: .github/CODEOWNERS
Comment (desaintmartin) [inline .github/CODEOWNERS:17]: Don't you want to maintain? ;)

File: charts/prometheus-elasticsearch-exporter/README.md
Comment (desaintmartin) [inline charts/prometheus-elasticsearch-exporter/README.md:12]:
```suggestion
- Kubernetes 1.10+
```

**Review by desaintmartin — COMMENTED** [review]
Review for elasticsearch-exporter only!

**Comment by torstenwalter** [comment]
I did not receive any feedback from @apenney for the stackdriver exporter chart. I would wait one or two days.

If I do not hear anything until then I would create a commit, which removes that charts. That way it would be easy to revert that commit to add the chart here. Any objections?

File: .github/CODEOWNERS
Comment (torstenwalter) [reply .github/CODEOWNERS:17]: I am just helping to migrate ;-)

**Comment by torstenwalter** [comment]
Hello @rpahli, @monotek mentioned in https://github.com/helm/charts/issues/23859#issuecomment-705102942 that you are also maintaining a stackdriver-exporter chart and that you could be interested in maintaining this one. Are you interested?

Do you know how I can fix the build without providing real google cloud credentials?

```
time="2020-10-15T19:17:05Z" level=info msg="Starting stackdriver_exporter (version=0.6.0, branch=HEAD, revision=59b7f38fa3d7e7ff8fd7134dd373c91051d5c16c)" source="stackdriver_exporter.go:121"
time="2020-10-15T19:17:05Z" level=info msg="Build context (go=go1.11.2, user=root@28c8704f4734, date=20181202-06:28:44)" source="stackdriver_exporter.go:122"
time="2020-10-15T19:17:05Z" level=error msg="Error creating Google client: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information." source="stackdriver_exporter.go:126"
```

File: charts/prometheus-stackdriver-exporter/README.md
Comment (scottrigby) [inline charts/prometheus-stackdriver-exporter/README.md:13]: The repo moved:
```suggestion
Learn more: <https://github.com/prometheus-community/stackdriver_exporter>
```

**Comment by rpahli** [comment]
@torstenwalter yes I'm intrested.

what you can try is to add a dummy credential as I remember the exporter need only a file to start:

```
{
   "type": "service_account",
   "project_id": "dummy-acc",
   "private_key_id": "00000000000000000000000000000000000",
   "private_key": "",
   "client_email": "dummy@dummy-acc.iam.gserviceaccount.com",
   "client_id": "0000000000000000000000000",
   "auth_uri": "https://accounts.google.com/o/oauth2/auth",
   "token_uri": "https://oauth2.googleapis.com/token",
   "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
   "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/dummy%dummy-acc.iam.gserviceaccount.com"
}
```

**Comment by torstenwalter** [comment]
@rpahli Thank you. Which email address should I use when adding you as maintainer?

**Comment by torstenwalter** [comment]
@prometheus-community/helm-charts-admins Please invite @rpahli to prometheus-community/helm-charts-maintainers

**Comment by rpahli** [comment]
my mail: pahli88@googlemail.com

**Comment by torstenwalter** [comment]
@rpahli Thank you to fix ci build. I added dummy credentials in [ci/default-values.yaml](https://github.com/prometheus-community/helm-charts/blob/f688aef22e3119c2adb045800737144a1bb2a615/charts/prometheus-stackdriver-exporter/ci/default-values.yaml)

@scottrigby We are good to go for another review round. I suggest that you approve CODEOWNER changes and one maintainer of each chart approves chart changes.
@rpahli that would include you as you are at the moment the only one for stackdriver-exporter :-) Thanks for your help!

**Review by rpahli — APPROVED** [review]
The stackdriver chart looks good to me

File: charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl
Comment (svenmueller) [inline charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl:5]: By best practice, the prefix for the named templates should be the chart name (which was changed during the migration process from `elasticsearch-exporter` to `prometheus-elasticsearch-exporter`)

> One popular naming convention is to prefix each defined template with the name of the chart: {{ define "mychart.labels" }}. By using the specific chart name as a prefix we can avoid any conflicts that may arise due to two different charts that implement templates of the same name.

(https://helm.sh/docs/chart_template_guide/named_templates/)

This would be incompatible with existing releases (since default release name includes the prefx). Any thoughts on this?

**Comment by svenmueller** [comment]
> Thanks for the initiative, of course i'm willing to continue maintaining it here!
> Note that I am already maintainer of other chart here so no need to add me to the org.

Same here! Thx for starting this initiative. I'm also happy to keep contributing 👍

File: charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl
Comment (torstenwalter) [reply charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl:5]: That's a really good point. I changed the name so that all charts in this repository use prometheus prefix.

I could mention in the docs that one could use `nameOverride` and `fullnameOverride` to keep the old names.

As this chart obviously has a new name I am not sure if we should increment a major version for this or just minor version. What's your view on that?

I also noticed that a container in a pod uses the chart name. That's shouldn't be a problem at all. I personally prefer to have a constant name and not to template that. That makes it easier to get logs from a specific container as I just have to remember that the container is named `exporter` no matter what the Deployment/StatefulSet is called.

**Comment by torstenwalter** [comment]
I checked my theory and setting name override and fullnameOverride seems to work

```console
$ helm install my-exporter stable/elasticsearch-exporter

$ kubectl get deploy
NAME                                 READY   UP-TO-DATE   AVAILABLE   AGE
my-exporter-elasticsearch-exporter   0/1     1            0           12s

$ kubectl get pod   
NAME                                                 READY   STATUS    RESTARTS   AGE
my-exporter-elasticsearch-exporter-546765858-62dv9   1/1     Running   0          19s

$ helm upgrade my-exporter . --set fullnameOverride=my-exporter-elasticsearch-exporter --set nameOverride=elasticsearch-exporter
Release "my-exporter" has been upgraded. Happy Helming!
NAME: my-exporter
LAST DEPLOYED: Wed Oct 21 21:47:56 2020
NAMESPACE: default
STATUS: deployed
REVISION: 2
TEST SUITE: None

$ helm list
NAME       	NAMESPACE	REVISION	UPDATED                                 	STATUS  	CHART                                  	APP VERSION
my-exporter	default  	2       	2020-10-21 21:47:56.229695404 +0200 CEST	deployed	prometheus-elasticsearch-exporter-3.8.0	1.1.0 

$ kubectl get deploy
NAME                                 READY   UP-TO-DATE   AVAILABLE   AGE
my-exporter-elasticsearch-exporter   1/1     1            1           58s

$ kubectl get pod   
NAME                                                  READY   STATUS    RESTARTS   AGE
my-exporter-elasticsearch-exporter-67b78b7c87-2lhlp   1/1     Running   0          28s
```

You can also verify the diff:

```
$ helm template my-exporter stable/elasticsearch-exporter > before.yam
$ helm template my-exporter . --set fullnameOverride=my-exporter-elasticsearch-exporter,nameOverride=elasticsearch-exporter > after.yaml
$ diff -C 1  before.yaml after.yaml
*** before.yaml	2020-10-21 21:52:30.094512517 +0200
--- after.yaml	2020-10-21 21:53:21.040668436 +0200
***************
*** 1,3 ****
  ---
! # Source: elasticsearch-exporter/templates/service.yaml
  kind: Service
--- 1,3 ----
  ---
! # Source: prometheus-elasticsearch-exporter/templates/service.yaml
  kind: Service
***************
*** 7,9 ****
    labels:
!     chart: elasticsearch-exporter-3.7.0
      app: elasticsearch-exporter
--- 7,9 ----
    labels:
!     chart: prometheus-elasticsearch-exporter-3.8.0
      app: elasticsearch-exporter
***************
*** 21,23 ****
  ---
! # Source: elasticsearch-exporter/templates/deployment.yaml
  apiVersion: apps/v1
--- 21,23 ----
  ---
! # Source: prometheus-elasticsearch-exporter/templates/deployment.yaml
  apiVersion: apps/v1
***************
*** 27,29 ****
    labels:
!     chart: elasticsearch-exporter-3.7.0
      app: elasticsearch-exporter
--- 27,29 ----
    labels:
!     chart: prometheus-elasticsearch-exporter-3.8.0
      app: elasticsearch-exporter
***************
*** 54,56 ****
        containers:
!         - name: elasticsearch-exporter
            env:
--- 54,56 ----
        containers:
!         - name: prometheus-elasticsearch-exporter
            env:


```

File: charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl
Comment (svenmueller) [reply charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl:5]: > As this chart obviously has a new name I am not sure if we should increment a major version for this or just minor version. What's your view on that?

- Since the user needs to adjust configuration (via `nameOverride`/`fullnameOverride`) in order to be able to re-use existing releases with the old naming scheme, i would vote for bumping the major version (incompatible change)
- Adding a migration section in the README would be very handy to have

> I also noticed that a container in a pod uses the chart name. That's shouldn't be a problem at all. I personally prefer to have a constant name and not to template that. That makes it easier to get logs from a specific container as I just have to remember that the container is named exporter no matter what the Deployment/StatefulSet is called.

I agree that container names can be hard coded values.

Comment (torstenwalter) [reply charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl:5]: @svenmueller bumped the major version now and added upgrade instructions. Could you please review?

**Comment by torstenwalter** [comment]
I hope that all issues are resolved now. I actually learned my lesson: Next time I migrate two more than one chart I will create once PR for each ;-)

For this we now need another round of approvals:
- 1 approval for stackdriver-exporter
- 1 approval for elasticserach-exporter
- 1 approval for CODEOWNER changes.

Hopefully everything is fine now ...

**Review by monotek — APPROVED** [review]

**Review by svenmueller — DISMISSED** [review]

File: charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl
Comment (svenmueller) [reply charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl:5]: Thx! The elasticsearch-exporter chart looks good to me 👍

**Review by rpahli — APPROVED** [review]

**Comment by torstenwalter** [comment]
Note: This should be merged using merge commit to preserve the history.

File: charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl
Comment (desaintmartin) [reply charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl:5]: I must be missing something: I still see all templates being prefixed by `elasticsearch-exporter` instead of `prometheus-elasticsearch-exporter`.

**Review by rpahli — APPROVED** [review]

**Review by svenmueller — APPROVED** [review]
LGTM

**Comment by desaintmartin** [comment]
What about my last note regarding name of templates? Should we change them?

**Comment by torstenwalter** [comment]
> What about my last note regarding name of templates? Should we change them?

Sorry I missed your comment (https://github.com/prometheus-community/helm-charts/pull/183/files#r511567862). Up to you I can rename those as part of this PR or it can be done in a follow up one. For users that should not make a difference as only the rendered manifests are of interest.

File: charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl
Comment (torstenwalter) [reply charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl:5]: @desaintmartin I don;t get you here. Where do you see those prefixes?

Most templates use something like `name: {{ template "elasticsearch-exporter.fullname" . }}-cert`

That template is defined like this:

```
{{/*
Create a default fully qualified app name.
We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
If release name contains chart name it will be used as a full name.
*/}}
{{- define "elasticsearch-exporter.fullname" -}}
{{- if .Values.fullnameOverride -}}
{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{- $name := default .Chart.Name .Values.nameOverride -}}
{{- if contains $name .Release.Name -}}
{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
{{- end -}}
{{- end -}}
{{- end -}}
```

So it should reflect the name of the chart unless you override it to keep the old name. Sure we could rename the template function, but that would not change the rendered content.

**Comment by torstenwalter** [comment]
I fixed DCO check by signing all commits. This required force pushing. Fingers crossed that I did not mess anything up while doing so.

**Review by monotek — APPROVED** [review]

**Review by scottrigby — APPROVED** [review]
👏

**Comment by monotek** [comment]
@scottrigby 
Please use your non squashing admin powers to merge preserving the old commits :)

**Comment by scottrigby** [comment]
k done, and merge commit option is disabled again.

**Comment by monotek** [comment]
@scottrigby 
Release job failed because of 

```
Run helm repo add stable https://kubernetes-charts.storage.googleapis.com/
Error: repo "https://kubernetes-charts.storage.googleapis.com/" is no longer available; try "https://charts.helm.sh/stable" instead
```

https://github.com/prometheus-community/helm-charts/runs/1310433064?check_suite_focus=true

**Comment by torstenwalter** [comment]
Was fixed with #260 and released by https://github.com/prometheus-community/helm-charts/runs/1310814662?check_suite_focus=true

Releases:
https://github.com/prometheus-community/helm-charts/releases/tag/prometheus-stackdriver-exporter-1.4.0
https://github.com/prometheus-community/helm-charts/releases/tag/prometheus-elasticsearch-exporter-4.0.0

**Pull request merged**
